### ISSUE TITLE
Punctuation in pdf in Exercise titles

### ISSUE THREAD

**Issue by nordstromjf** [rbeezer/mathbook#926]
I am using [] in exercise titles in my PreTeXt code as follows:

  <exercise xml:id="E_LPBF">
    <title>Expected value of [B, F]</title>
    <statement>
      <p>
        Similarly, determine the expected value for Player 1 for the pure strategy pair [B, F]. What is Player 2's expected value?
      </p>
    </statement>

The HTML puts the period after the ] in the title and looks fine, but the pdf generated from the latex version has [B, F .] (space, period, ])

**Comment by rbeezer** [comment]
Ooh, that's bad.  Likely the brackets are being confused with the end of an 
optional argument.

It should be fixed.  But in your case, is "[B,F]" mathematics?  Are you using an 
"m" tag?  If not, is it better that way?

Maybe the title needs "protection" with curly braces, like I believe is 
happening with optional short titles already (migrating to the ToC).

**Comment by davidfarmer** [comment]
The part should be in math mode, but there could be some case
we did not think of yet that has regular square brackets.

**Comment by nordstromjf** [comment]
The first instance I have this notation in a title, I would like it to say “Payoff for [Bet, Fold].” With the syntax I posted in the issue, the pdf says “Payoff for [Bet, Fold .]” The html is fine, and looks like I want. Here are some things I tried and what happened:

<title>Payoff for <m>[Bet, Fold]</m></title>, of course the Bet and Fold are italicized in the html, but the big problem is I couldn’t tex it since when I ran pdflatex it got an error: missing $.

<m>[</m>Bet,Fold<m>]</m>, and <m>[\text{Bet}, \text{Fold}]</m>, here the html still looks a little weird because the math font is different from the regular font, but again I get the missing $ error with pdflatex.

I think it is the use of [] that is causing problems, since if I try <m>Bet, Fold</m> with no [ ], I don’t get an error.

I already switched notation throughout the book since {Bet, Fold} had problems in the pdf (missing {}).

So, basically the use of [] in math tags makes things worse in titles.

Jen
------------------------------------------
Jennifer Firkins Nordstrom
Department of Mathematics
Linfield College

jfirkins@linfield.edu

**Comment by rbeezer** [comment]
Thanks for the debugging.  The use of [,] and {,} by LaTeX for grouping and 
arguments is almost certainly the root cause, along with titles moving around.

We've protected against this sort of thing befor - I've got an idea for a fix.

You could try the PTX elements, <braces> and <brackets> (iirc)for now, as well. 
Not sure if they will behave better.  They might.

**Comment by nordstromjf** [comment]
Hi Rob,

The <braces> element works just like [ ]. I have updated both my html and my matching pdf on my git hub page. So if you are curious you can see how the pdf looks: https://nordstromjf.github.io/IntroGameTheoryv4.pdf

Section 2.4 has the first occurrence.

Again, the HTML looks just fine.


Jen
------------------------------------------
Jennifer Firkins Nordstrom
Department of Mathematics
Linfield College

jfirkins@linfield.edu

**Comment by rbeezer** [comment]
Thanks for the testing.  I'll get it fixed.  Very busy with other PTX work for two weeks.  What is your timeline for a beautiful PDF?

**Comment by nordstromjf** [comment]
No rush. I posted what I have. Unlikely that I will repost any time soon.

Thanks,

Jen

Sent from my iPad

On Jul 30, 2018, at 12:32 PM, Rob Beezer <notifications@github.com<mailto:notifications@github.com>> wrote:

Thanks for the testing. I'll get it fixed. Very busy with other PTX work for two weeks. What is your timeline for a beautiful PDF?

**Comment by rbeezer** [comment]
This should be fixed now.  Thanks for your patience waiting for this.  Testing with your examples would be appreciated.

Checkpoint 25.19 (or 25.20) of the sample article (which is now exposing an existing numbering bug).

Should fix about 10 of the similar blocks that are being styled via `tcolorbox` now.  Fix at 46fa86197c8a685cfeb4520bf590613a2b91a218.

Rather than protecting optional arguments with groups (curly braces), I'd like to replace dangerous characters by safe equivalents, such as `\textlbrace`.  But I can't seem to find one for square brackets.

**Comment by rbeezer** [comment]
And I should have also said:  thanks for reporting this and testing the bad behavior.
